exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -247,6 +247,75 @@ class EntityIDColumnType<T : Comparable<T>>(
     override fun hashCode(): Int = 31 * super.hashCode() + idColumn.hashCode()
 }
 
+/**
+ * An interface defining the transformation between a database column type and a real type.
+ *
+ * @param TReal The real type to which the column value is transformed.
+ * @param TColumn The column type provided by underlying `ColumnType`.
+ */
+interface ColumnTransformer<TReal, TColumn> {
+    /**
+     * Transforms a column value from the underlying `ColumnType` into the real type.
+     */
+    fun toReal(value: TColumn): TReal
+
+    /**
+     * Transforms a real type value into a column value.
+     */
+    fun toColumn(value: TReal): TColumn
+}
+
+class ColumnTransformerImpl<TReal, TColumn>(
```

Comment:
please consider making it private

Reply:
It's used in both: DSL transform, and DAO transform which are in different packages. I could copy it and make private in both packages.

Reply:
or it can be `internal`

Reply:
It's the first thing I tried to do, but it is not reachable from the `exposed-dao` module..

Reply:
Another variant is in ktorm: 
```kotlin
fun <C : Any, R : Any> Column<C>.transform(
    fromUnderlyingValue: (C) -> R,
    toUnderlyingValue: (R) -> C
): Column<R>
```

Hibernate also has similar concept, but it applies modification directly to the sql, and looks like the following annotation:
```java
@ColumnTransformer(
  read="decrypt(credit_card_num)" 
  write="encrypt(?)"
)
String creditCardNumber;
```

In SqlDelight there is a similar to our transform thing called `ColumnAdapter`:
```kotlin
object : ColumnAdapter<List<String>, String> {
  override fun decode(databaseValue: String) =
    if (databaseValue.isEmpty()) listOf() else databaseValue.split(",")
  override fun encode(value: List<String>) = value.joinToString(separator = ",")
}
```